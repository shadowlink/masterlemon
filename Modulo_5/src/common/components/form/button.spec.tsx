import * as React from 'react';
import { shallow } from 'enzyme';
import { Button } from './button';

describe('common/components/form/button specs', () => {
  it('should render as expected when passing required properties', () => {
    // Arrange
    const props = {
      label: 'test label',
      onClick: () => { },
    };

    // Act
    const component = shallow(
      <Button
        {...props}
      />
    );

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should render as expected when passing required and optional properties', () => {
    // Arrange
    const props = {
      label: 'test label',
      onClick: () => { },
      type: 'test type',
    };

    // Act
    const component = shallow(
      <Button
        {...props}
      />
    );

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should call onClick prop when simulate button click', () => {
    // Arrange
    const props = {
      label: 'test label',
      onClick: jest.fn(),
      type: 'test type',
    };

    // Act
    const component = shallow(
      <Button
        {...props}
      />
    );

    const preventDefaultSpy = jest.fn();
    component.simulate('click', {
      preventDefault: preventDefaultSpy,
    });

    // Assert
    expect(props.onClick).toHaveBeenCalled();
    expect(preventDefaultSpy).toHaveBeenCalled();
  });
});
