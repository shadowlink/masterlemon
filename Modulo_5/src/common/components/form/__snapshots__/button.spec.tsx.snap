// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/form/button specs should render as expected when passing required and optional properties 1`] = `
<button
  className="btn btn-lg btn-success btn-block"
  onClick={[Function]}
  type="test type"
>
  test label
</button>
`;

exports[`common/components/form/button specs should render as expected when passing required properties 1`] = `
<button
  className="btn btn-lg btn-success btn-block"
  onClick={[Function]}
  type="submit"
>
  test label
</button>
`;
